config/config.go

```diff
@@ -25,6 +25,58 @@ import (
 	tracing "github.com/uber/jaeger-client-go/config"
 )
 
+// Config number limitations
+const (
+	MinPortNumber            = 1
+	MaxPortNumber            = 65535
+	MinTokenLimit            = 1
+	MaxTokenLimit            = 2000
+	MinLogSize               = 1024
+	MaxLogSize               = 1 << 32
+	MinLogDays               = 1
+	MaxLogDays               = 2 * 365
+	MinBackups               = 1
+	MaxBackups               = 1000
+	MinSlowLogThreshold      = 100
+	MaxSlowLogThreshold      = 300 * 1000
+	MinExpensiveLogThreshold = 10
+	MaxExpensiveLogThreshold = 1 << 20
+	MinQueryLogLength        = 100
+	MaxQueryLogLength        = 10000
+	MinMetricsInterval       = 1
+	MaxMetricsInterval       = 120
+	MinProcs                 = 0
+	MaxProcs                 = 1000
+	MinRetryLimit            = 0
+	MaxRetryLimit            = 100
+	MinJoinConcurrency       = 1
+	MaxJoinConcurrency       = 100
+	MinStmtCountLimit        = 1000
+	MaxStmtCountLimit        = 1 << 20
+	MinPlanCacheCap          = 1
+	MaxPlanCacheCap          = 10000
+	MinPlanCacheShards       = 1
+	MaxPlanCacheShards       = 1000
+	MinPrePlanCacheCap       = 1
+	MaxPrePlanCacheCap       = 1000
+	MinProxyHeaderTimeout    = 1
+	MaxProxyHeaderTimeout    = 120
+	MinGrpcConnCount         = 1
+	MaxGrpcConnCount         = 1000
+)
+
+// Valid config maps
+var (
+	ValidStorage = map[string]bool{
+		"mocktikv": true,
+		"tikv":     true,
+	}
+	// TODO: Remove it when X Server is available.
+	ValidXServer = map[bool]bool{
```

Comment:
Why not just use a bool value `false`?

Reply:
In my opinion, there are two kinds of limitations for settings. One is the range like, and the other is the list like. I prefer to use a uniform style for the two styles.